### ISSUE TITLE
When loading dataset from libsvm file, determine size of line

### ISSUE THREAD

**Issue by jdrzj** [yoshoku/rumale#27]
Hi,
In libsvm [MNIST file](https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass/mnist.t.bz2) there are skipped zero values, and it makes size of one line not correct.
My suggestion here is to add option to determine size of libsvm line.

```ruby
2.7.1 :004 > x, y = Rumale::Dataset.load_libsvm_file("mnist", zero_based: true)

2.7.1 :005 > 
2.7.1 :006 > y
 => 
Numo::Int32#shape=[60000]
[5, 0, 4, 1, 9, 2, 1, 3, 1, 4, 3, 5, 3, 6, 1, 7, 2, 8, 6, 9, 4, 0, 9, 1, 1, ...] 
2.7.1 :007 > x
 => 
Numo::DFloat#shape=[60000,781]
```

My desired size of x is `[60000,784]`.

If any knows how can I do it, I'd be happy to hear that.

**Comment by kojix2** [comment]
Do we need something like the max_index option?

**Comment by yoshoku** [comment]
@jdrzj 
Thank you for your request. I have added a new parameter `n_features` to load_libsvm_file for specifying the number of features, and have released 0.22.5 including that.

```ruby
$ irb
irb(main):001:0' require 'rumale'
=> true
irb(main):002:0> x, y = Rumale::Dataset.load_libsvm_file('mnist.t', n_features: 780)
=>
[Numo::DFloat#shape=[10000,780]
...
irb(main):003:0>
```

https://yoshoku.github.io/rumale/doc/Rumale/Dataset.html#load_libsvm_file-class_method

**Comment by jdrzj** [comment]
Thank you @yoshoku for looking on this and changing this. 🙌
